crates/rustc_codegen_spirv/src/abi.rs

```diff
@@ -818,6 +818,25 @@ fn trans_image<'tcx>(
             }
             Some(SpirvType::Sampler.def(span, cx))
         }
+        SpirvAttribute::SampledImage => {
+            // see SpirvType::sizeof
+            if ty.size != Size::from_bytes(4) {
+                cx.tcx.sess.err("#[spirv(sampled_image)] type must have size 4");
+                return None;
+            }
+
+            // We use a generic to indicate the underlying image type of the sampled image.
+            // The spirv type of it will be generated by querying the type of the first generic.
+            if let TyKind::Adt(_, substs) = *ty.ty.kind() {
+                // TODO: enforce that the generic param is an image type?
+                let image_ty = substs.type_at(0);
```

Comment:
Nit: I think this might ICE if it's not a generic type (as in doesn't have a type parameter 0), but considering `#[spirv(sampled_image)]` is only intended to be used in `spirv-std`, that's probably okay.